dotnet-client-libraries/ITextBasics/ITextBasics/BigDocument.cs

```diff
@@ -0,0 +1,87 @@
+using iText.Layout.Properties;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace ITextBasics
+{
+    public class BigDocument
+    {
+        public void Create(string pdfFileName)
+        {
+            using var writer = new PdfWriter(pdfFileName);
+            using var pdfDocument = new PdfDocument(writer);
+            using var document = new Document(pdfDocument);
+
+            string? executablePath = Path.GetDirectoryName(Assembly.GetEntryAssembly()!.Location);
+            string imageFile = Path.Combine(executablePath!, "Resources", "Flower.jpg");
+
+            for (int i = 0; i < 3; i++)
+            {
+                document.Add(CreateHeader1(LoremIpsum(3)));
+                document.Add(CreateHeader2(LoremIpsum(7)));
+                document.Add(CreateNormalParagraph(LoremIpsum(100)));
+
+                document.Add(CreateHeader2(LoremIpsum(7)));
+                document.Add(CreateNormalParagraph(LoremIpsum(55)));
+                document.Add(CreateNormalParagraph(LoremIpsum(27)));
+
+                document.Add(CreateHeader1(LoremIpsum(6)));
+                document.Add(CreateHeader2(LoremIpsum(3)));
+                document.Add(CreateNormalParagraph(LoremIpsum(50)));
+                document.Add(CreatePictureParagraph(imageFile));
+                document.Add(CreateHeader2(LoremIpsum(3)));
+                document.Add(CreateNormalParagraph(LoremIpsum(45)));
+            }
+        }
+
+        private Paragraph CreateHeader1(string caption)
+        {
+            return new Paragraph(caption)
+                .SetTextAlignment(TextAlignment.CENTER)
+                .SetFontSize(20)
+                .SetBold();
+        }
+
+        private Paragraph CreateHeader2(string caption)
+        {
+            return new Paragraph(caption)
+                .SetTextAlignment(TextAlignment.LEFT)
+                .SetFontSize(15)
+                .SetBold();
+        }
+
+        private Paragraph CreateNormalParagraph(string text)
+        {
+            return new Paragraph(text);
+        }
+
+        private Paragraph CreatePictureParagraph(string imageFile)
+        {
+            var imageData = ImageDataFactory.Create(imageFile);
+            var image = new Image(imageData);
+
+            return new Paragraph()
+                .Add(image);
+        }
+
+        private string LoremIpsum(int numberOfWords)
+        {
+            var loremIpsumText = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.";
+            var words = loremIpsumText.Split(' ');
+
+            var sentences = new List<string>();

```

Comment:
This could be done with a StringBuilder and then calls to AppendJoin. We could also create the StringBuilder with a starting capacity based on the number of words. Maybe something like numberOfWords * 10 or instead of 10 we could use the average length of the Ipsum words.

Reply:
HI,

In the light of my previous answer. If this is something I really have to do, no problem, of course, I can do it. But again! Realy? Should we do that in the code that is meant for beginners? What will anyone gain from that?

Reply:
Following on to my answer on the previous point, our goal is to provide code that follows best practices. In this case, rewriting this code with a StringBuilder will make the code shorter, easier to follow and follow best practices.

Reply:
I used your code, so it is now using StringBuilder